### ISSUE TITLE
Core Entity ::create() is ineffective with constructor parameters

### ISSUE THREAD

**Issue by PWesterdale** [reservat/core#22]
https://github.com/reservat/core/blob/master/src/Core/Entity.php#L12

Does not take into account constructor parameters on `new static()`
This means it will fail when it does not supply the correct amount of arguments.
Only solution is to remove constructor arguments, or have `= false` on each.

Not sure what to do with this :/

**Comment by PWesterdale** [comment]
Three options:

```
function __construct(...$params){
   $this->field = $params[0];
   ....
}
```

```
function __construct($data){
   // data must be associative array of property => data
   return static::create($data);
}
```

```
function __construct($name = null, $email = null){
   $this->name = $name;
   $this->email = $email;
}
```

**Comment by PWesterdale** [comment]
New Solution:

I've implemented this in Session/Auth/Core for testing purposes.

```

// Customer.php

function __construct($name, $email){} // Normal Constructor

static function createFromArray(array $data)
{
   return new static($data['name'], $data['email']);
}

/* 
CustomerManager.php
We use this mostly internally, When we know what we want we would usually 
call new Customer(), but inside the application it may need to create entities, 
and when doing this would usually create from data in an array (i.e From the DB)
*/

function getEntity(...$args){
   return Customer::createFromArray(...$args);
}

/* PDORepository Base Class

function getResults()
{
   /* 
  This function now just returns an array of entity arrays or a single entity array, 
  which you can then pass to getEntity() or createFromArray()
  */
}

```

Hope that makes sense?
It's a pretty big breaking change but I'm noticing that its a fair bit faster than the magic of create just from doing tests. (tests were taking 3 seconds, now under 1 second).
It means we can keep standard constructors, but does mean we have to maintain a createFromArray function on each entity, but I don't feel thats a lot of bother?

:heart:

**Comment by PWesterdale** [comment]
Fixed in #27

**Comment by PWesterdale** [comment]
CLERSED
